### PR TITLE
Refactored SentinelGrpcServerInterceptorTest

### PR DESCRIPTION
**Pull request by refactormachine**
Performed extract class operation for SentinelGrpcServerInterceptorTest. Extracted fields: server. Extracted Methods: prepareServer, stopServer.

### CODE REVIEW

File: sentinel-adapter/sentinel-grpc-adapter/src/test/java/com/alibaba/csp/sentinel/adapter/grpc/GrpcTestServer.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import static org.junit.Assert.*;
+
+class GrpcTestServer {
+    private Server server;
+
+    GrpcTestServer() {
+    }
+
+    void prepare(int port) throws IOException {
```
Comment (sczyh30) [inline sentinel-adapter/sentinel-grpc-adapter/src/test/java/com/alibaba/csp/sentinel/adapter/grpc/GrpcTestServer.java]: Using the name `start` may be more appropriate.

File: sentinel-adapter/sentinel-grpc-adapter/src/test/java/com/alibaba/csp/sentinel/adapter/grpc/GrpcTestServer.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package com.alibaba.csp.sentinel.adapter.grpc;
+
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+
+import java.io.IOException;
+
+import static org.junit.Assert.*;
```
Comment (sczyh30) [inline sentinel-adapter/sentinel-grpc-adapter/src/test/java/com/alibaba/csp/sentinel/adapter/grpc/GrpcTestServer.java]: You can remove unused imports.

File: sentinel-adapter/sentinel-grpc-adapter/src/test/java/com/alibaba/csp/sentinel/adapter/grpc/SentinelGrpcServerInterceptorTest.java [hunk 3]
```diff
@@ -43,8 +39,8 @@
 
     private final String resourceName = "com.alibaba.sentinel.examples.FooService/anotherHello";
     private final int threshold = 4;
+    private final GrpcTestServer extractedSentinelGrpcServerInterceptorTest = new GrpcTestServer();
```
Comment (sczyh30) [inline sentinel-adapter/sentinel-grpc-adapter/src/test/java/com/alibaba/csp/sentinel/adapter/grpc/SentinelGrpcServerInterceptorTest.java]: The name is too long, just `grpcTestServer` is okay.

File: sentinel-adapter/sentinel-grpc-adapter/src/test/java/com/alibaba/csp/sentinel/adapter/grpc/GrpcTestServer.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    GrpcTestServer() {
+    }
+
+    void start(int port, boolean shouldintercept) throws IOException {
+        if (server != null) {
+            throw new IllegalStateException("Server already running!");
+        }
+        ServerBuilder<?> serverBuild = ServerBuilder.forPort(port)
+                .addService(new FooServiceImpl());
+        if(shouldintercept)
```
Comment (sczyh30) [inline sentinel-adapter/sentinel-grpc-adapter/src/test/java/com/alibaba/csp/sentinel/adapter/grpc/GrpcTestServer.java]: Please reformat your code with [Alibaba P3C Plugin](https://github.com/alibaba/p3c) and obey the rules in Alibaba Java Coding Guidelines (naming, indent, format, etc.). For example:

```java
if (shouldIntercept) {
    serverBuilder.intercept(new SentinelGrpcServerInterceptor());
}
```

**Comment by refactormachine** [comment]
Hi @sczyh30 , reformatted the code.

**Comment by sczyh30** [comment]
Nice, thanks for your contribution!

**Pull request merged**